### ISSUE TITLE
Form field should not have multiple label elements

### ISSUE THREAD

**Issue by nicolad** [chakra-ui/chakra-ui#2756]
# 🐛 Bug report

While testing `Switch` component for `a11y` with `axe` chrome extension, following error was showing

![image](https://user-images.githubusercontent.com/1477453/101800270-c62b2380-3b15-11eb-9a68-08f2e74d95c8.png)

***
It would be nice to provide a way to use built-in `label` optionally.
For example, in a case when `Switch` is used with a label related with `aria-describedby` we shouldn't have that wrapping label.

This issue happens when `Switch` is used inside `FormControl` along with `FormLabel`.

**Comment by manzoorwanijk** [comment]
Also, `<Switch />` and `<Checkbox />` should probably have `role="checkbox"` by default and also have `aria-checked={isChecked}`.

**Comment by with-heart** [comment]
> Also, `<Switch />` and `<Checkbox />` should probably have `role="checkbox"` by default and also have `aria-checked={isChecked}`.

I'll test again to verify, but I think we should be fine on that front as we render a hidden native `input type="checkbox"` element in both cases. The label is associated to the hidden `input` and the rendered checkbox/switch is just a custom visual that matches the state of the hidden `input`.

> The first rule of ARIA is: if a native HTML element or attribute has the semantics and behavior you require, use it instead of re-purposing an element and adding an ARIA role, state or property to make it accessible. As such, it is recommended to use the native HTML checkbox  using form control instead of recreating a checkbox's functionality with JavaScript and ARIA.

Source: "Best Practices" section of https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/checkbox_role

https://www.sarasoueidan.com/blog/inclusively-hiding-and-styling-checkboxes-and-radio-buttons/ has a little more.
